Sources/TensorFlow/Core/LazyTensorOperation.swift

```diff
@@ -725,3 +724,70 @@ extension LazyTensorOperation: CustomStringConvertible {
         return desc
     }
 }
+
+extension LazyTensorOperation {
+    /// Returns the materialized value at the given output `index`.
+    func materialized(index: Int) -> TFETensorHandle {
+        precondition(index < outputCount)
+        return materialized()[index]
+    }
+
+    /// Materializes all the outputs.
+    func materialized() -> [TFETensorHandle] {
+        // Return materialized outputs if any.
+        if let outputs = outputs { return outputs }
+
+        LazyTensorOperation.materializeLiveTensors(self)
+
+        // Our outputs should have been updated by now. Otherwise,
+        // something terrible happened!
+        precondition(outputs != nil, "Materialization failed!")
+        return outputs!
+    }
+
+    /// Converts symbolic tensor inputs to concrete inputs if the
+    /// associated `LazyTensorOperation` has been materialized.
+    private func maybeMaterializeInputs() {
+        func maybeMaterialized(lazyTensor: LazyTensor) -> LazyTensor {
+            let handle = lazyTensor.handle
+            if case let LazyTensor.Handle.symbolic(lazyOp, index, _) = handle {
+                if let outputs = lazyOp.outputs {
+                    return LazyTensor(_materialized: outputs[index])
+                }
+            }
+            return lazyTensor
+        }
+
+        func maybeMaterialized(input: Input) -> Input {
```

Comment:
Could you elaborate on what "maybe" stands for here?

Reply:
Added a brief comment to the function. Basically, if any symbolic value in the input has been materialized, the input is rewritten by replacing the symbolic value with the corresponding concrete value.

Reply:
How about `materializedAsNeeded`?

Reply:
SG. Done!